config/config.py

```diff
@@ -1,187 +1,330 @@
-import yaml, os
+import yaml
+import os
 
-class _DeskConfigDAO:
-	"""
-	Desk Config DAO
-	"""
 
-	KEY_ID = "id"
-	KEY_NAME = "name"
-	KEY_SENSOR_PIN = "sensor_pin"
+class _SensorTypeConfigDAO:
+    """
+    Sensor Type Config DAO
+    """
 
-	def __init__(self, desk):
-		self.__desk_id = desk[self.KEY_ID]
-		self.__desk_name = desk[self.KEY_NAME]
-		self.__desk_pin = desk[self.KEY_SENSOR_PIN]
+    KEY_TYPE = "type"
+    KEY_NAME = "name"
 
-	@property
-	def id(self):
-		return self.__desk_id
+    def __init__(self, sensor):
+        self.__sensor_type = sensor[self.KEY_TYPE]
+        self.__name = sensor[self.KEY_NAME]
 
-	@property
-	def name(self):
-		return self.__desk_name
+    @property
+    def sensor_type(self):
+        return self.__sensor_type
 
-	@property
-	def pin(self):
-		return self.__desk_pin
+    @property
+    def sensor_name(self):
+        return self.__name
 
+    @property
+    def properties(self):
+        return {self.KEY_TYPE: self.sensor_type, self.KEY_NAME: self.__name}
 
-class _DebugConfigDAO:
-	"""
-	Debug Config DAO
-	"""
+    def __str__(self):
+        return "Sensor:\n\tType: " + str(self.__sensor_type) + "\n\tName: " + str(self.__name)
 
-	KEY_ENABLED = "enabled"
-	KEY_REMOTE_SIMULATOR_CONFIG = "remote-simulator"
 
-	def __init__(self, debug):
-		self.__enabled = debug[self.KEY_ENABLED]
-		self.__remote_simulator_config = _RemoteSimulatorConfigDAO(debug[self.KEY_REMOTE_SIMULATOR_CONFIG])
+class _DeskSensorConfigDAO:
+    """
+    Desk Sensor Config DAO
+    """
 
-	@property
-	def enabled(self):
-		return self.__enabled
+    KEY_TYPE = "type"
+    KEY_BUS = "i2c_bus"
+    KEY_ADDRESS = "i2c_address"
 
-	@property
-	def simulator_config(self):
-		return self.__remote_simulator_config
+    def __init__(self, desk):
+        self.__sensor_type = desk[self.KEY_TYPE]
+        self.__i2c_bus = desk[self.KEY_BUS]
+        if type(desk[self.KEY_ADDRESS]) is int:
+            self.__i2c_address = desk[self.KEY_ADDRESS]
+        else:
+            self.__i2c_address = int(desk[self.KEY_ADDRESS], 16)
 
+    @property
+    def sensor_type(self):
+        return self.__sensor_type
 
-class _RemoteSimulatorConfigDAO:
-	"""
-	Simulator Config DAO
-	"""
-	KEY_ENABLED = "enabled"
-	KEY_PORT = "port"
+    @property
+    def i2c_bus(self):
+        return self.__i2c_bus
 
-	def __init__(self, simulator):
-		self.__enabled = simulator[self.KEY_ENABLED]
-		self.__port = simulator[self.KEY_PORT]
+    @property
+    def i2c_address(self):
+        return self.__i2c_address
 
-	@property
-	def enabled(self):
-		return self.__enabled
+    @sensor_type.setter
```

Comment:
Do we need all these setters?

Reply:
Not at the moment, but I was kind of future-proofing in a way, since I was already fixing the config stuff. This way we can modify the config via CLI, if needed, and it could be easily saved to file. If it ends up truly being useless it wouldn't be hard to remove.